live-build/misc/migration-scripts/dx_apply

```diff
@@ -16,9 +16,190 @@
 #
 
 #
-# [WIP]
-# The existence of this file is checked by dx_unpack, thus we provide
-# it as an empty file until the needed functionality is added.
+# Creates the dataset layout/hierarchy expected by linux-upgrade for the
+# archive from the migration image to be unpacked. It also takes the
+# required steps and adds an entry to the FreeBSD bootloader so we can
+# optionally boot into Linux.
 #
 
+set -o pipefail
+
+function die() {
+	echo "$(basename "$0"): $*" >&2
+	exit 1
+}
+
+function usage() {
+	echo "$(basename "$0"): $*" >&2
+	echo "Usage: $(basename "$0") <archive-directory>"
+	exit 2
+}
+
+[[ "$(uname -s)" == "SunOS" ]] || die "script can only be used in illumos"
+
+while getopts ':h' opt; do
+	case $opt in
+	h) usage ;;
+	*) usage "Invalid option: -$OPTARG." ;;
+	esac
+done
+
+ARCHIVE_DIR=$1
+[[ -n $ARCHIVE_DIR ]] || usage
+[[ -d $ARCHIVE_DIR ]] || die "$ARCHIVE_DIR is not a directory"
+[[ -f $ARCHIVE_DIR/version.info ]] ||
+	die "$ARCHIVE_DIR does not have a version.info file"
+. "$ARCHIVE_DIR/version.info"
+#
+# WIP: Assert MINIMUM VERSION field for migration.
```

Comment:
Yes, so there are basically 2 checks about the minimum version.

The first one happens right after we source `version.info` from the migration archive:
```
 52 . "$ARCHIVE_DIR/version.info"
 53 #
 54 # WIP: Assert MINIMUM VERSION field for migration.
 55 #
```
This is to ensure that the archive itself provides the correct minimum version. The idea is to guard against this migration script used against some random image that is for example just upgrade and not migration.

The second one is looking for what is currently installed on the system, and making sure that the system is at that minimum version:
```
 74 #
 75 # Check that the version to which we are upgrading is compatible with the
 76 # currently installed version. (Look at version.info to see if we are
 77 # above the minimum version required. Since we are doing a migration
 78 # the minimum version should be the latest 5.3.
 79 #
 80 # CURRENT_DDS=$(dirname "$(mount | awk '/^\/opt\/delphix /{ print $3 }')")
 81 # CURRENT_VERSION=$(basename $current_dds)
 82 #
 83 # WIP: check if current version is 5.3, if not bail
 84 #
```

TL;DR - First check is to trying to check if we have the right archive. Second check ensures that the system is at the minimum version.

Does this sound reasonable? If yes, I can write more comments to clarify. If not, I can just take the comments out of there.